src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/SearchController.php

```diff
@@ -65,4 +70,46 @@ public function indexAction(Request $request, AliasFilters $filters): Response
             ],
         ]);
     }
+
+    /**
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function searchAliasesForAssociationAction(Request $request): JsonResponse
+    {
+        try {
+            /** @var SearchTerm[] $searchTerms */
+            $searchTerms = $this->getQueryBus()->handle(new SearchForSearchTerm(
+                $request->get('query', ''),
+                (int) $request->get('limit', SearchForSearchTerm::DEFAULT_LIMIT)
+            ));
+        } catch (AliasConstraintException $e) {
+            return $this->json([
+                'message' => $this->getErrorMessage($e),
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        if (empty($searchTerms)) {
+            return $this->json(['aliases' => []], Response::HTTP_NOT_FOUND);
```

Comment:
This action should return array of search terms by query needed for `searchTerm` input as it needs to display all existing searchTerms in database by this query. If there is an existing search term in this array it means that the merchant will edit existing aliases.

@zuk3975 WDYT about renaming it to `SearchForAliases` instead?

Reply:
I think we already discussed this, you already renamed it from "searchAliases" to "SearchSearchterms" because that what it does (searches for terms and not aliases). You just forgot to rename here in controller.
return $this->json(['searchTerms' => []], Response::HTTP_NOT_FOUND);